ansible_base/authentication/models/social_auth.py

```diff
@@ -1,10 +1,25 @@
+from base64 import b64encode
+from typing import Any
+
 from django.conf import settings
 from django.db import models
 from social_django.models import AbstractUserSocialAuth
 
 from ansible_base.authentication.models import Authenticator
 
 
+def b64_encode_binary_data_in_dict(obj: Any) -> Any:
```

Comment:
This concerns me a little in that we're not tracking which fields are getting base64-encoded. So we have no way to ever revert the process, because we don't know which fields to decode.

I was originally going to ask down below where you call this in `save()` if we need a way to revert the process on object load - but there's no way to do that with this implementation. Could that come back to bite us?

Reply:
This field is the extra data that we get from an authentication. For example, it might look like:
```json
{
    "displayname": ["User Name"],
    "uid": ["username"],
    "phone_number": "123-456-7890",
    "country": "Germany",
    ....
    "auth_time": "2024-04-17T15:45:56.066473Z"
}
```

We reference this data when processing `attribute` authenticator_maps. So if there is a binary blob in the extra data, like maybe a GPG key, whoever is setting up the authenticator_map would just need to know to use the b64 version of the key. To me that seems like a pretty rare use case so I don't think there is really a need for us to be able to un-convert the field.

But I'm open to other opinions. We can't really change the field type because its probvide by social_auth but we could add a second text field with a copy but again, kind of an edge case to begin with IMHO.